### PR TITLE
Added support an intra-cell-newlines; fix #3

### LINKED ISSUE #3: Does not properly handle cells with newlines in them
This extension does not properly handle Excel cells that have newlines in them. Here is some CSV that illustrates the issue. The issue presents itself with "Tax Rates" in the Label column.

Label,Content
Tax City Code,002
Tax City Name,Ada
Tax City Type,Village
Tax County Names,Hardin
Tax State Name,Ohio
Member Type,Full
FIPS Code,39-00198
Population,"5,952"
Postal Codes,45810
Income Cutoff,">= $5,000"
Growth Rate,3.0%
"Tax Rates
    2002-2011
    2011-2018","
1.15%
1.65%"
"Penalty Rate
    Minimum
    Maximum","1.5% monthly
$25 once
none"
Interest Rate,1.5% monthly
Tax Lottery,yes
Mandatory Filing,yes

### PR DESCRIPTION
**Pull request by usagi**
#3

### CODE REVIEW

File: src/excel-markdown-helpers.ts [hunk 1]
```diff
@@ -5,12 +5,17 @@ const ALIGNED_LEFT = "l";
 const ALIGNED_RIGHT = "r";
 const ALIGNED_CENTER = "c";
 const EXCEL_COLUMN_DELIMITER = "\t";
+const MARKDOWN_INTRA_CELL_NEWLINE = "<br/>";
+const UNESCAPED_DOUBLE_QUTE = '"';
```
Comment (csholmq) [inline src/excel-markdown-helpers.ts]: QUOTE*

**Comment by usagi** [comment]
@csholmq ping

File: src/excel-markdown-helpers.ts [hunk 2]
```diff
@@ -5,12 +5,17 @@ const ALIGNED_LEFT = "l";
 const ALIGNED_RIGHT = "r";
 const ALIGNED_CENTER = "c";
 const EXCEL_COLUMN_DELIMITER = "\t";
+const MARKDOWN_INTRA_CELL_NEWLINE = "<br/>";
```
Comment (csholmq) [inline src/excel-markdown-helpers.ts]: Maybe drop INTRA_CELL in the defines for better readability.

**Review by csholmq — CHANGES_REQUESTED** [review]
Looking very nice!

Which platforms have you tested this on?

**Comment by csholmq** [comment]
@usagi I want to merge this but there are two stoppers.

1. Conflicting file `src/test/excel-markdown-helpers.test.ts`
2. Unable to debug this. Seems to have broken since the last time I managed this. I assume #14 fixes this?

**Comment by usagi** [comment]
`rebase` is completed. 🚀

**Comment by csholmq** [comment]
@usagi Doesn't appear to work for me on Win10.

https://docs.google.com/spreadsheets/d/1RHlfdDmaGdCHUuT7rYlAWnuo8zoiP8EHYcAH4zRM4bg/edit?usp=sharing

produces

```
| aaa  | bbb  |
|------|------|
| ccc  | "ddd |
| eee" |
```

**Comment by csholmq** [comment]
Oh, ok. Found the issue. First execution works fine, but the second is bad. Reason seems to be some kind of escaping when replacing the copy buffer with the original raw data. Something is changed.

**Comment by usagi** [comment]
Hmm, I only tested from Microsoft Excel, not from Google Spreadsheet. And It's work fine from Microsoft Excel.

image

**Comment by csholmq** [comment]
First execution

`"aaa\tbbb\r\nccc\t"ddd\neee""`

Second execution

`"aaa\tbbb\r\nccc\t"ddd\r\neee""`

**Comment by usagi** [comment]
> Oh, ok. Found the issue. First execution works fine, but the second is bad. Reason seems to be some kind of escaping when replacing the copy buffer with the original raw data. Something is changed.

Good finding, I could repro too.

**Comment by usagi** [comment]
Ok, I'll fix it.

**Comment by csholmq** [comment]
Might be an issue with https://github.com/xavi-/node-copy-paste so perhaps difficult to workaround.

If you could also drop the `INTRA_CELL` in the other defines as well and then squash your commits.

**Comment by usagi** [comment]
> Might be an issue with https://github.com/xavi-/node-copy-paste so perhaps difficult to workaround.

No problems maybe. I have an idea. Plz wait the next fixing.

**Comment by usagi** [comment]
rebased with:
1. `INTRA_CELL` dropping
2. fixup the pasting issue. -> see src/extension.ts https://github.com/csholmq/vscode-excel-to-markdown-table/pull/15/files#diff-45327f86d4438556066de133327f4ca2R44

**Comment by csholmq** [comment]
Nice work! Thanks!

I'll add you to the credits and release a new version tonight. Many thanks! 🎉💯

**Pull request merged**